### PR TITLE
Clean up stored procedure tests as NonSharedModelTestBase

### PR DESCRIPTION
**Pull request by roji**
This is a test-only change which redoes the sproc tests as NonSharedModelTestBase, making them much less brittle and encapsulating each tests requirements inside the test. Since these tests are new, I'd like to get this in for 7.0 if possible, so that external providers can use this new model and not have to do another change later.

### CODE REVIEW

**Review by ajcvickers — APPROVED** [review]

File: test/EFCore.Specification.Tests/TestUtilities/TestHelpers.cs [hunk 1]
```diff
@@ -238,10 +238,7 @@ public static void AssertAllMethodsOverridden(Type testClass)
             .GetRuntimeMethods()
             .Where(
                 m => m.DeclaringType != testClass
-                    && m.GetCustomAttributes()
-                        .Any(
-                            a => a is ConditionalFactAttribute
-                                || a is ConditionalTheoryAttribute))
+                    && m.GetCustomAttributes().Any(a => a is ConditionalFactAttribute or ConditionalTheoryAttribute))
```
Comment (AndriySvyryd) [inline test/EFCore.Specification.Tests/TestUtilities/TestHelpers.cs:241]: Using `Attribute.IsDefined` would likely be (slightly) faster

File: test/EFCore.Relational.Specification.Tests/Update/StoredProcedureUpdateTestBase.cs [hunk 2]
```diff
@@ -560,89 +901,206 @@ public virtual async Task Tph(bool async)
...
         }
     }
 
-    private async Task SaveChanges(StoredProcedureUpdateContext context, bool async)
+    /// <summary>
+    ///     A method to be implement by the provider, to set up a store-generated concurrency token shadow property with the given name.
+    /// </summary>
+    protected abstract void ConfigureStoreGeneratedConcurrencyToken(EntityTypeBuilder entityTypeBuilder, string propertyName);
+
+    public class Entity
```
Comment (AndriySvyryd) [inline test/EFCore.Relational.Specification.Tests/Update/StoredProcedureUpdateTestBase.cs:1036]: Make all these protected

**Review by AndriySvyryd — APPROVED** [review]

File: test/EFCore.Relational.Specification.Tests/Update/StoredProcedureUpdateTestBase.cs [hunk 3]
```diff
@@ -560,89 +901,198 @@ public virtual async Task Tph(bool async)
...
 
     protected virtual void ClearLog()
-        => Fixture.TestSqlLoggerFactory.Clear();
+        => TestSqlLoggerFactory.Clear();
 
-    protected TFixture Fixture { get; }
+    protected virtual void CreateStoredProcedures(DbContext context, string createSprocSql)
+    {
+        foreach (var batch in
+                 new Regex("^GO", RegexOptions.IgnoreCase | RegexOptions.Multiline, TimeSpan.FromMilliseconds(1000.0))
```
Comment (danmoseley) [inline test/EFCore.Relational.Specification.Tests/Update/StoredProcedureUpdateTestBase.cs:1092]: Feel free to ignore, but regex timeouts are a protection against DOS attacks - I assume input here is trusted. Given that test machines are occasionally perversely slow, I wonder whether you want a timeout here, or if a timeout then a large one like 1 min.

Comment (roji) [reply test/EFCore.Relational.Specification.Tests/Update/StoredProcedureUpdateTestBase.cs:1092]: Yeah, this is test-only code which we already use elsewhere, and haven't ever seen timeouts AFAIK... We can definitely crank it up if this happens (or just not use regex for this trivial thing).

**Pull request merged**